### PR TITLE
feat(bom): Generate and write changelog.

### PR DESCRIPTION
**Pull request by jtk54**
This generates the changelog using `clog-cli` and writes it to a file. I'll add the changelog publication step soon.

### CODE REVIEW

File: dev/generate_bom.py [hunk 1]
```diff
@@ -125,6 +127,38 @@ def write_docker_version_files(self):
...
+        print "Changelog generation failed for {0} with \n{1}\n exiting...".format(comp, result.stdout)
+        exit(result.returncode)
+      # Capitalize
+      comp_cap = comp[0].upper() + comp[1:]
+      changelog += '\n# {0}\n{1}'.format(comp_cap, result.stdout)
+    print 'Writing changelog...'
+    # Write the changelog with the toplevel version without the build number.
+    # This is ok since the changelog is only published if the toplevel version is released.
+    changelog_file = '{0}-changelog.md'.format(self.__toplevel_version)
+    with open(changelog_file, 'w') as clog:
```
Comment (ewiseblatt) [inline dev/generate_bom.py]: You shoudl be able to control where the output goes other than CWD.

File: dev/generate_bom.py [hunk 2]
```diff
@@ -125,6 +127,38 @@ def write_docker_version_files(self):
...
         cfg.write(docker_tag)
 
+  def generate_changelog(self):
+    """Generate a release changelog and write it to a file.
+
+    The changelog contains a section per microservice that describes the
+    changes made since the last Spinnaker release. It also contains the
+    version information as well.
+    """
+    changelog = 'Spinnaker {0}\n\n'.format(self.__toplevel_version)
```
Comment (ewiseblatt) [inline dev/generate_bom.py]: changelog shoudl be a [] that you append lines to.
I guess you can append a blank.
You should probably informally sign it with a date and tool so when someone has it, they know where it came from.

File: dev/generate_bom.py [hunk 3]
```diff
@@ -125,6 +127,38 @@ def write_docker_version_files(self):
...
+    for comp, hash in self.__changelog_start_hashes.iteritems():
+      version_bump = self.__component_versions[comp]
+      next_tag_with_build = '{0}-{1}'.format(version_bump.version_str,
+                                             self.build_number)
+      first_dash_idx = next_tag_with_build.index('-')
+      gradle_version = next_tag_with_build[first_dash_idx + 1:]
```
Comment (ewiseblatt) [inline dev/generate_bom.py]: where did gradle come from? These are from git tags, right?

File: dev/generate_bom.py (hunk 1 above)
Comment (jtk54) [inline dev/generate_bom.py]: Do you mean specify the file or the output path with the file name as I have it now?

**Comment by jtk54** [comment]
Added option for changelog output file, cleaned some things up -- merging if no complaints.

**Pull request merged**